### ISSUE TITLE
TC rule (flake8-typechecking) fails to apply on imports containing both runtime and typehinting items

### ISSUE THREAD

**Issue by rsokolewicz** [astral-sh/ruff#28060]
### Summary

I noticed that if I have something like

```
from .quz import Foo, Bar

def baz() -> Foo:
    return Bar()
```

that the TC rule doesn't trigger, whereas if I have

```
from .quz import Foo
from .quz import Bar

def baz() -> Foo:
    return Bar()
```

The TC rule does trigger, and suggests to move Foo inside a TYPE_CHECKING block

```
from typing import TYPE_CHECKING

from .quz import Bar

if TYPE_CHECKING:
    from .quz import Foo

def baz() -> Foo:
    return Bar()
```

### Version

ruff 0.16.4

**Comment by Daverball** [comment]
See [lint.flake8-type-checking.strict](https://docs.astral.sh/ruff/settings/#lint_flake8-type-checking_strict).

This is intentional and matches the default behavior of the original flake8 plugin. If you want imports to always be flagged set `strict` to `true`.

Your examples also don't quite reflect the truth, it doesn't matter if the imports are split into multiple statements or originate from a single statement. All that matters is whether they're all coming from the same module.

**Comment by rsokolewicz** [comment]
okay, thanks for clearing that up :) closing.
